File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        arr=set(nums)
        longest=0
        for n in arr:
            if (n-1) not in arr:
                length=0
                while (n+length) in arr:
                    length+=1
                longest=max(length,longest)
        return longest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length==0) return 0;

        Set<Integer> set = new HashSet<>();
        for(int num: nums){
            set.add(num);
        }
        int maxCount = 0;
        
        for(Integer num: set){
            
            if(!set.contains(num-1)){
                int count = 1;
                int currentNum = num;
                while(set.contains(currentNum+1)){
                    count++;
                    currentNum+=1;
                }
                maxCount = Math.max(maxCount, count);
            }
        }
        return maxCount;
    }
}